yupiik-batch-runtime/src/main/java/io/yupiik/batch/runtime/batch/builder/BatchChain.java

```diff
@@ -104,7 +104,12 @@ default void run(final RunConfiguration configuration) { // todo: shouldn't we f
                     if (result.value() instanceof BatchPromise<?> promise) {
                         final var end = promise.end().toCompletableFuture();
                         promises.add(end);
-                        end.whenComplete((ok, ko) -> promises.remove(end));
+//                        end.whenComplete((ok, ko) -> promises.remove(end));
+                        end.whenComplete((ok, ko) -> {
+                            if (!end.isCompletedExceptionally() || configuration == null || !configuration.failOnPromiseError) {
```

Comment:
maybe `forceAwaitOnPromiseError` since we already fail if it does not finish early enough?